statusHandler/persistor/persistentHandler.go

```diff
@@ -0,0 +1,209 @@
+package persistor
+
+import (
+	"sync"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/logger"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	"github.com/ElrondNetwork/elrond-go/storage/storageUnit"
+)
+
+const saveInterval = 10 * time.Second
+
+//StatusMetricsDbEntry is key used to save metrics in storage
+const StatusMetricsDbEntry = "status_metrics_db_entry"
+
+var log = logger.DefaultLogger()
+
+// PersistentStatusHandler is a status handler that will save metrics in storage
+type PersistentStatusHandler struct {
+	store             storage.Storer
+	persistentMetrics *sync.Map
+	marshalizer       marshal.Marshalizer
+}
+
+// NewPersistentStatusHandler will return an instance of the persistent status handler
+func NewPersistentStatusHandler(
+	marshalizer marshal.Marshalizer,
+) (*PersistentStatusHandler, error) {
+	if marshalizer == nil || marshalizer.IsInterfaceNil() {
+		return nil, process.ErrNilMarshalizer
+	}
+
+	psh := new(PersistentStatusHandler)
+	psh.store = storageUnit.NewNilStorer()
```

Comment:
I don't think you should use a NilStorer here. Those are used just as placeholders.
The storer should come from the constructor.

Reply:
Yea you right NilStorer it is a placeholder because i cannot use a storer component when PersistentStatusHandler is created because in main.go first time is created coreComponents that contains status handlers and after that are created dataComponents that contains storer. And i can use a storer just after dataComponents are created.